calcom/cal.com
# feat: Implements maxLeadThreshold

which puts a maximum to the amount of lead a given host is allowed to be ahead

<img width="952" alt="image" src="https://github.com/user-attachments/assets/6d060d07-b50a-45a7-a599-511e6f582fde">

## What does this PR do?

Implements a fairness algorithm

✅ ~TODO: 99% finished handleNewBooking algorithm part (reuses most logic here)~
🚫 ~TODO: E2E test~ Opted against this in favour of vitest, considering the logic is covered by other e2e sufficiently.
✅  ~TODO: Allow setting the maxLeadThreshold from the UI (in seperate branch atm)~

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?

* See tests in vitest for filterHostsByLeadThreshold

Manual path:

1. Using the seeded team, ensure two users are assigned to a RR event type - `user 1` & `user 2`.
2. Give `user 1` extra availability, such as Saturday.
3. Book Saturday repeatedly for 3 times, the 4th time `user 1` will be disabled.
4. Book `user 2`
5. Saturday will open up again as an option as `user 1` is re-enabled.
